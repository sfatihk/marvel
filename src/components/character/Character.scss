@import "../../assets/style/mixin.scss";

.character {
  @include characterCoverFrame;
  background-color: #fff;

  display: inline-block;
  height: 300px;
  margin: 1vmin;
  overflow: hidden;
  position: relative;
  width: 300px;

  cursor: pointer;
}

.character img {
  transform-origin: 100 100;
  transition: transform 0.5s ease;
  flex-shrink: 0;
  min-height: 100%;
  min-width: 100%;
  max-height: 100%;
}

.character:hover img {
  transform: scale(1.1);
}

.character__name {
  background-color: #fff;
  border: solid 2px #000;
  margin: 0;
  padding: 3px 10px;
  font-size: 18px;
  bottom: -2px;
  position: absolute;
  right: -6px;
  transform: skew(-15deg);
}
