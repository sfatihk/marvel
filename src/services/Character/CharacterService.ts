import FetchCharactersRequest, {
  FetchCharactersPayloadType,
} from "./requests/FetchCharactersRequest";

import fetchCharacterDetailsRequest, {
  FetchCharacterDetailsPayloadType,
} from "./requests/FetchCharacterDetailRequest";

import FetchCharacterComicsRequest, {
  FetchCharacterComicsPayloadType,
} from "./requests/FetchCharacterComicsRequest";

const fetchCharacters = (props: FetchCharactersPayloadType) =>
  FetchCharactersRequest(props);

const fetchCharacterDetails = (props: FetchCharacterDetailsPayloadType) =>
  fetchCharacterDetailsRequest(props);

const fetchCharacterComics = (props: FetchCharacterComicsPayloadType) =>
  FetchCharacterComicsRequest(props);

export { fetchCharacters, fetchCharacterDetails, fetchCharacterComics };
