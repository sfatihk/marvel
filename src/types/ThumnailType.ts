export type ThumnailType = {
  extension: string;
  path: string;
};
