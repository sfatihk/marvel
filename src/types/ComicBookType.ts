import { ThumnailType } from "./ThumnailType";

type UrlType = {
  type: "detail" | "purchase" | "reader" | "inAppLink";
  url: string;
};
type DateType = {
  type: "onsaleDate" | "focDate" | "unlimitedDate" | "digitalPurchaseDate";
  date: string;
};
type PriceType = {
  type: "printPrice" | "digitalPurchasePrice";
  price: number;
};
export type ComicBookType = {
  name: string;
};

export type ExtendedComicBookType = {
  id: number;
  title: string;
  pageCount: number;
  urls: UrlType[];
  dates: DateType[];
  prices: PriceType[];
  thumbnail: ThumnailType;
};
