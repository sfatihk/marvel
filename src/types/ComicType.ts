import { ComicBookType } from "./ComicBookType";
export type ComicType = {
  items: ComicBookType[];
};
