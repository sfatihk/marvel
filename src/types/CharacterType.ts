import { ThumnailType } from "./ThumnailType";
import { ComicType } from "./ComicType";

export type CharacterType = {
  id: number;
  name: string;
  thumbnail: ThumnailType;
  description: string;
  comics: ComicType;
};
