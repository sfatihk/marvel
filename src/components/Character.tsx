import { CharacterType } from "../types/CharacterType";
import { useHistory } from "react-router-dom";

//styles
import "./Character.scss";
const Character = ({
  id,
  name,
  description,
  thumbnail,
  comics,
}: CharacterType) => {
  const history = useHistory();
  return (
    <div
      onClick={() => history.push(`/characters/${id}`)}
      className="character"
    >
      <img
        src={thumbnail.path + "." + thumbnail.extension}
        alt=""
        className="character__thumbnail"
      />
      <div className="character__name">{name}</div>
    </div>
  );
};

export default Character;
