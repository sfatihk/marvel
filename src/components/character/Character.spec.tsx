import renderer from "react-test-renderer";
import { CharacterType } from "../../types/CharacterType";
import Character from "./Character";

describe("Character components", () => {
  const initialState: CharacterType = {
    id: 1,
    name: "test",
    thumbnail: { extension: "png", path: "xyz" },
    comics: [],
    description: "test desctiption",
  };

  it("renders correctly", () => {
    const tree = renderer.create(<Character {...initialState} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
