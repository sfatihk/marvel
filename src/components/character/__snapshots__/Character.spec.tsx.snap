// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character components renders correctly 1`] = `
<div
  className="character"
  onClick={[Function]}
>
  <img
    alt=""
    className="character__thumbnail"
    src="xyz.png"
  />
  <div
    className="character__name"
  >
    test
  </div>
</div>
`;
